packages/@aws-cdk/aws-servicediscovery/lib/non-ip-instance.ts

```diff
@@ -37,7 +36,7 @@ export class NonIpInstance extends InstanceBase {
   constructor(scope: Construct, id: string, props: NonIpInstanceProps) {
     super(scope, id);
 
-    if (props.service.namespace.type !== NamespaceType.HTTP) {
+    if (props.service.discoveryType !== DiscoveryType.API) {
```

Comment:
I'm concerned that if somebody has created a namespace with type `NamespaceType.HTTP`, this check doesn't work anymore. If I understand you correctly, you're saying this isn't possible with the provided constructs. However the contract seems to very much allow this. E.g. someone could have created a custom implementation.

Reply:
Ahh yes I only considered the use-case where the services are created by CDK; I will fix this up

Reply:
👍🏻 Thanks for addressing this. I haven't checked the rest of the code yet, but the same issue might be present with other added conditionals.

Reply:
I think the other conditionals are safer as they are all part of the Service implementation and based on the existing required parameters passed through

Reply:
I added an additional guard here which protects against creating the service without using the namespace helpers